### PR TITLE
{devel}[dummy,GCC/5.2.0] Boost 1.59.0 with full icu support

### PR DESCRIPTION
**Pull request by torbjoernk**
The 'parallel' option is matching an addition in the Boost easyblock (see hpcugent/easybuild-easyblocks#701).

Also had to add an update for the bzip2 config.

[Review on Reviewable](https://reviewable.io/reviews/hpcugent/easybuild-easyconfigs/1996)

### CODE REVIEW

File: easybuild/easyconfigs/b/bzip2/bzip2-1.0.6-GCC-5.2.0.eb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+toolchain = {'name': 'GCC', 'version': '5.2.0'}
+toolchainopts = {'pic': True}
+
+sources = [SOURCE_TAR_GZ]
+source_urls = ['http://www.bzip.org/%(version)s/']
+
+import os
+os.environ['CC'] = 'gcc'
+os.environ['CFLAGS'] = '-O3 -fPIC'
```
Comment (wpoely86) [inline easybuild/easyconfigs/b/bzip2/bzip2-1.0.6-GCC-5.2.0.eb]: this is done by EB itself. It should not be done in the easyconfig (run with the `-l` flag to see the variables being set).

File: easybuild/easyconfigs/b/Boost/Boost-1.59.0-GCC-5.2.0.eb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+boost_mpi = False
+
+configopts = '--without-libraries=python --with-icu'
+
+toolset = 'gcc'
+
+osdependencies = [('zlib-devel', 'zlib1g-dev'), ('libicu-devel', 'icu-devel', 'libicu-dev')]
+
+parallel = 8
```
Comment (wpoely86) [inline easybuild/easyconfigs/b/Boost/Boost-1.59.0-GCC-5.2.0.eb]: EB will by default utilize the maximal number of cores in the current cpuset/cgroup. It you want to limit this to maximally 8, use `maxparallel`. Else this line can go.

File: easybuild/easyconfigs/b/Boost/Boost-1.59.0-GCC-5.2.0.eb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+dependencies = [('bzip2', '1.0.6')]
+
+boost_mpi = False
+
+configopts = '--without-libraries=python --with-icu'
+
+toolset = 'gcc'
+
+osdependencies = [('zlib-devel', 'zlib1g-dev'), ('libicu-devel', 'icu-devel', 'libicu-dev')]
```
Comment (wpoely86) [inline easybuild/easyconfigs/b/Boost/Boost-1.59.0-GCC-5.2.0.eb]: split this line, 1 dep per line (for example, look in the other easyconfigs for Boost).

File: easybuild/easyconfigs/b/Boost/Boost-1.59.0-GCC-5.2.0.eb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+source_urls = [SOURCEFORGE_SOURCE]
+sources = ['%%(namelower)s_%s.tar.gz' % '_'.join(version.split('.'))]
+
+dependencies = [('bzip2', '1.0.6')]
+
+boost_mpi = False
+
+configopts = '--without-libraries=python --with-icu'
+
+toolset = 'gcc'
```
Comment (wpoely86) [inline easybuild/easyconfigs/b/Boost/Boost-1.59.0-GCC-5.2.0.eb]: why should this do?

File: easybuild/easyconfigs/b/bzip2/bzip2-1.0.6-GCC-5.2.0.eb (hunk 1 above)
Comment (torbjoernk) [inline easybuild/easyconfigs/b/bzip2/bzip2-1.0.6-GCC-5.2.0.eb]: I've added that bzip2 config as EB was complaining that it couldn't resolve the bzip2 dependency when I ran `eb Boost-1.59.0-GCC-5.2.0.eb -r`. Thus I've just copied `bzip2-1.0.6-GCC-4.9.2.eb` and adjusted GCC to 5.2.0.

File: easybuild/easyconfigs/b/Boost/Boost-1.59.0-GCC-5.2.0.eb [hunk 5]
```diff
@@ -0,0 +1,25 @@
+name = 'Boost'
+version = '1.59.0'
+
+homepage = 'http://www.boost.org/'
+description = """Boost provides free peer-reviewed portable C++ source libraries."""
+
+toolchain = {'name': 'GCC', 'version': '5.2.0'}
```
Comment (wpoely86) [inline easybuild/easyconfigs/b/Boost/Boost-1.59.0-GCC-5.2.0.eb]: We don't use the GCC toolchain anymore directly. Use `GNU` which is a combination of GCC with binutils. There reason for this is that RHEL 6 has a binutils that is too old to support Haswell hardware.

**Comment by wpoely86** [comment]
ok to test

File: easybuild/easyconfigs/b/Boost/Boost-1.59.0-GCC-5.2.0.eb (hunk 4 above)
Comment (torbjoernk) [inline easybuild/easyconfigs/b/Boost/Boost-1.59.0-GCC-5.2.0.eb]: `toolset` is an option from the Boost easyblock for specifying the so called _toolset_ for b2.

File: easybuild/easyconfigs/b/Boost/Boost-1.59.0-GCC-5.2.0.eb (hunk 5 above)
Comment (torbjoernk) [inline easybuild/easyconfigs/b/Boost/Boost-1.59.0-GCC-5.2.0.eb]: Didn't knew that. Going to change that.

File: easybuild/easyconfigs/b/bzip2/bzip2-1.0.6-GCC-5.2.0.eb (hunk 1 above)
Comment (wpoely86) [inline easybuild/easyconfigs/b/bzip2/bzip2-1.0.6-GCC-5.2.0.eb]: This should only be present in the `dummy` bzip2 easyconfig. 

But this easyconfig can go if you switch to the `GNU` toolchain. It already has a bzip2 easyconfig.

File: easybuild/easyconfigs/b/Boost/Boost-1.59.0-GCC-5.2.0.eb (hunk 3 above)
Comment (boegel) [inline easybuild/easyconfigs/b/Boost/Boost-1.59.0-GCC-5.2.0.eb]: why not list `zlib` and `libicu` as EasyBuild dependencies?

**Comment by boegel** [comment]
@torbjoernk: first things first: welcome to EasyBuild! Let us know if something isn't clear to you. Or maybe @ocaisa or @geimer can help or answer your questions. ;-)

**Comment by torbjoernk** [comment]
Thanks guys. I'll rework this PR a little more ontop of #2001 and try to get libicu into EB.

**Pull request closed without merging**